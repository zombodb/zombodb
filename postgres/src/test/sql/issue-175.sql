CREATE TABLE city (
  id        BIGSERIAL NOT NULL PRIMARY KEY,
  city_name VARCHAR(255)
);

INSERT INTO city (city_name) VALUES ('Seattle');
INSERT INTO city (city_name) VALUES ('Washington');

CREATE TABLE client (
  id         BIGSERIAL NOT NULL PRIMARY KEY,
  first_name VARCHAR(255),
  last_name  VARCHAR(255),
  city_id    BIGINT    NOT NULL REFERENCES city (id)
);

INSERT INTO client (first_name, last_name, city_id)
VALUES ('John', 'Smith', 1);

INSERT INTO client (first_name, last_name, city_id)
VALUES ('John', 'Doe', 1);

INSERT INTO client (first_name, last_name, city_id)
VALUES ('John', 'Wick', 2);

CREATE INDEX idxcity
  ON city USING zombodb(zdb('city', city.ctid), zdb(city)) WITH (url='http://localhost:9200/');
CREATE INDEX idxclient
  ON client USING zombodb(zdb('client', client.ctid), zdb(client)) WITH (url='http://localhost:9200/');

ALTER INDEX idxcity SET ( OPTIONS = 'id=<client.idxclient>city_id');
ALTER INDEX idxclient SET ( OPTIONS = 'city_id=<city.idxcity>id');

CREATE VIEW find_me AS
  SELECT
    client.*,
    city.city_name,
    zdb('client', client.ctid) AS zdb
  FROM client
    JOIN city ON client.city_id = city.id;

select * from find_me where zdb ==> 'john' order by id;

select * from find_me where zdb ==> 'seattle' order by id;

select * from find_me where zdb ==> 'john smith' order by id;

select * from find_me where zdb ==> 'smith seattle' order by id;

DROP TABLE city CASCADE;
DROP TABLE client CASCADE;
