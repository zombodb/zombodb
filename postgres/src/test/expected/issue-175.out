CREATE TABLE city (
  id        BIGSERIAL NOT NULL PRIMARY KEY,
  city_name VARCHAR(255)
);
INSERT INTO city (city_name) VALUES ('Seattle');
INSERT INTO city (city_name) VALUES ('Washington');
CREATE TABLE client (
  id         BIGSERIAL NOT NULL PRIMARY KEY,
  first_name VARCHAR(255),
  last_name  VARCHAR(255),
  city_id    BIGINT    NOT NULL REFERENCES city (id)
);
INSERT INTO client (first_name, last_name, city_id)
VALUES ('John', 'Smith', 1);
INSERT INTO client (first_name, last_name, city_id)
VALUES ('John', 'Doe', 1);
INSERT INTO client (first_name, last_name, city_id)
VALUES ('John', 'Wick', 2);
CREATE INDEX idxcity
  ON city USING zombodb(zdb('city', city.ctid), zdb(city)) WITH (url='http://localhost:9200/');
CREATE INDEX idxclient
  ON client USING zombodb(zdb('client', client.ctid), zdb(client)) WITH (url='http://localhost:9200/');
ALTER INDEX idxcity SET ( OPTIONS = 'id=<client.idxclient>city_id');
ALTER INDEX idxclient SET ( OPTIONS = 'city_id=<city.idxcity>id');
CREATE VIEW find_me AS
  SELECT
    client.*,
    city.city_name,
    zdb('client', client.ctid) AS zdb
  FROM client
    JOIN city ON client.city_id = city.id;
select * from find_me where zdb ==> 'john' order by id;
 id | first_name | last_name | city_id | city_name  |  zdb  
----+------------+-----------+---------+------------+-------
  1 | John       | Smith     |       1 | Seattle    | (0,1)
  2 | John       | Doe       |       1 | Seattle    | (0,2)
  3 | John       | Wick      |       2 | Washington | (0,3)
(3 rows)

select * from find_me where zdb ==> 'seattle' order by id;
 id | first_name | last_name | city_id | city_name |  zdb  
----+------------+-----------+---------+-----------+-------
  1 | John       | Smith     |       1 | Seattle   | (0,1)
  2 | John       | Doe       |       1 | Seattle   | (0,2)
(2 rows)

select * from find_me where zdb ==> 'john smith' order by id;
 id | first_name | last_name | city_id | city_name |  zdb  
----+------------+-----------+---------+-----------+-------
  1 | John       | Smith     |       1 | Seattle   | (0,1)
(1 row)

select * from find_me where zdb ==> 'smith seattle' order by id;
 id | first_name | last_name | city_id | city_name |  zdb  
----+------------+-----------+---------+-----------+-------
  1 | John       | Smith     |       1 | Seattle   | (0,1)
(1 row)

DROP TABLE city CASCADE;
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to constraint client_city_id_fkey on table client
drop cascades to view find_me
DROP TABLE client CASCADE;
